File: ddmlib/src/main/java/com/android/ddmlib/input/android/InputReader.java
package com.android.ddmlib.input.android;

import com.android.ddmlib.Log;

import java.util.HashMap;

/**
 * Created by majipeng on 2017/6/19.
 */
public class InputReader {

    HashMap<String, EventMapper> mappers = new HashMap<>();

    KnownEventList knownEventList = new KnownEventList();
    private EventHub eventHub;


    public InputReader(EventHub eventHub) {
        this.eventHub = eventHub;
    }

    /**
     * 非异步
     */
    public MonitorEvent readBySync() {
        try {
            RawEvent rawEvent = eventHub.getEvent();
            EventMapper mapper = mappers.get(rawEvent.getDevFile());
            if (mapper == null) {
                mapper = new EventMapperImpl(knownEventList);
                mappers.put(rawEvent.getDevFile(), mapper);
            }
            return mapper.processEvent(rawEvent);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
        return null;
    }


}

File: ddmlib/src/main/java/com/android/ddmlib/input/android/InputReader.kt
package com.android.ddmlib.input.android

import java.util.HashMap

/**
 * Created by majipeng on 2017/6/19.
 */
class InputReader internal constructor(private val eventHub: EventHub) {

    internal var mappers = HashMap<String, EventMapper>()

    internal var knownEventList = KnownEventList()

    /**
     * 非异步
     */
    fun readBySync(): MonitorEvent? {
        try {
            val rawEvent = eventHub.event
            var mapper: EventMapper? = mappers[rawEvent.devFile]
            if (mapper == null) {
                mapper = EventMapperImpl(knownEventList)
                mappers.put(rawEvent.devFile, mapper)
            }
            return mapper.processEvent(rawEvent)
        } catch (e: InterruptedException) {
            e.printStackTrace()
        }

        return null
    }


}
